compiler/rustc_session/src/session.rs

```diff
@@ -1009,6 +1010,12 @@ fn default_emitter(
                 );
                 Box::new(emitter.ui_testing(sopts.unstable_opts.ui_testing))
             } else {
+                let metrics = sopts.unstable_opts.error_metrics.as_ref().and_then(|path| {
+                    let mut path = path.clone();
+                    std::fs::create_dir_all(&path).ok()?;
+                    path.push(format!("error_metrics_{}", std::process::id()));
```

Comment:
Why PID? Can we use something more ordered such as `YYYY-MM-DD-HH-SS-PID`?

Reply:
I am not sure of the precise reasoning behind using PID, but I have no objections to switching to datetime-pid. We can definitely use that instead.